src/wallet/rpcwallet.cpp

```diff
@@ -4601,7 +4691,11 @@ UniValue z_mergetoaddress(const UniValue& params, bool fHelp)
         // Get available notes
         std::vector<SproutNoteEntry> sproutEntries;
         std::vector<SaplingNoteEntry> saplingEntries;
-        pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, zaddrs);
+        std::optional<AddrSet> noteFilter =
+            useAnySprout || useAnySapling ?
+                std::nullopt :
```

Comment:
This will cause us to select all Sprout and Sapling notes if either `useAnySprout` or `useAnySapling` is `true`. This however doesn't cause any issues, because below here we throw an error if we would try to spend both Sprout and Sapling notes in the same merge transaction.

Reply:
I'm confused as to whether this is changing the behaviour?

Reply:
I think that it is not changing the behavior, because the `zaddrs` set previously passed to GetFilteredNotes was forced to be empty by the conditions on the previous lines 4453-58. This would have resulted in both `sproutEntries` and `saplingEntries` being populated. So this retains the existing behavior; the existing behavior could very reasonably be considered suboptimal. A good refactor would be to change `AddrSet` to be able to directly reflect the "all" cases internally, and that could get rid of some of the extra complexity here. A good time to do that will be when refactoring z_mergetoaddress to also use the transaction builder.